lib/api/api_auth_test.go

```diff
@@ -65,3 +65,63 @@ func TestAuthLDAPSendsCorrectBindDNWithNoTemplate(t *testing.T) {
 		t.Fatalf("ldapTemplateBindDN should be %s != %s", expectedDn, templatedDn)
 	}
 }
+
+func TestNoAuthPathsIsImmutable(t *testing.T) {
+	// It is an important security assumption that noAuthPaths is immutable, but
```

Comment:
What's the presumed attack behind this important security assumption? It seems to me like this is something we don't do anywhere else.